pkg/serverless/proc/proc.go

```diff
@@ -114,3 +121,36 @@ func GetCPUData(path string) (float64, float64, error) {
 
 	return userCPUTimeMs, systemCPUTimeMs, nil
 }
+
+type NetworkData struct {
+	RxBytes float64
+	TxBytes float64
+}
+
+// GetNetworkData collects bytes sent and received by the function
+func GetNetworkData(path string) (*NetworkData, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	var interfaceName string
+	var rxBytes, rxPackets, rxErrs, rxDrop, rxFifo, rxFrame, rxCompressed, rxMulticast, txBytes,
+		txPackets, txErrs, txDrop, txFifo, txColls, txCarrier, txCompressed float64
+	for {
+		_, err = fmt.Fscanln(file, &interfaceName, &rxBytes, &rxPackets, &rxErrs, &rxDrop, &rxFifo, &rxFrame,
+			&rxCompressed, &rxMulticast, &txBytes, &txPackets, &txErrs, &txDrop, &txFifo, &txColls, &txCarrier,
+			&txCompressed)
+		if errors.Is(err, io.EOF) {
+			return nil, err
```

Comment:
You can consider returning a new error instance here. Since an EOF error is not very descriptive of the problem, returning an error like "network data not found in file xyz" can be more helpful.

Reply:
Good call 👍